<?xml version="1.0"?>
<launch>
            
            <!-- arguments -->
            <arg name="domain_path" default="$(find erl2)/PDDL/domain.pddl" />
            <arg name="problem_path" default="$(find erl2)/PDDL/problem.pddl" />
            
            <!-- knowledge base -->
            <node name="rosplan_knowledge_base" pkg="rosplan_knowledge_base" type="knowledgeBase" respawn="false" output="screen">
            <param name="domain_path" value="$(arg domain_path)" />
            
            <param name="problem_path" value="$(arg problem_path)" />
                    <!-- conditional planning flags -->
                    <param name="use_unknowns" value="false" />
            </node>
            
            <!-- problem generation -->
            <include file="$(find rosplan_planning_system)/launch/includes/problem_interface.launch">
                    <arg name="knowledge_base" value="rosplan_knowledge_base" />
                    <arg name="domain_path" value="$(arg domain_path)" />
                    <arg name="problem_path" value="$(find erl2)/PDDL/new_problem.pddl" />
                    <arg name="problem_topic" value="problem_instance" />
            </include>
            
            <!-- planner interface -->
            <include file="$(find rosplan_planning_system)/launch/includes/planner_interface.launch">
                    <arg name="use_problem_topic" value="true" />
                    <arg name="problem_topic" value="/rosplan_problem_interface/problem_instance" />
                    <arg name="planner_topic" value="planner_output" />
                    <arg name="problem_path"  value="$(find erl2)/PDDL/problem.pddl" />
                    <arg name="domain_path" value="$(arg domain_path)" />
                    <arg name="data_path" value="$(find erl2)/PDDL/" />
                    <arg name="planner_command" value="timeout 10 $(find rosplan_planning_system)/common/bin/popf DOMAIN PROBLEM" />
            </include>
            
             <!-- plan parsing -->
            <node name="rosplan_parsing_interface" pkg="rosplan_planning_system" type="pddl_simple_plan_parser" respawn="false" output="screen">
            <param name="knowledge_base" value="rosplan_knowledge_base" />
            <param name="planner_topic" value="/rosplan_planner_interface/planner_output" />
            <param name="plan_topic" value="complete_plan" />
            </node>
            
             <!-- plan dispatching --> 
            <node name="rosplan_plan_dispatcher" pkg="rosplan_planning_system" type="pddl_simple_plan_dispatcher" respawn="false" output="screen">
            <param name="knowledge_base" value="rosplan_knowledge_base" />
            <param name="plan_topic" value="/rosplan_parsing_interface/complete_plan" />
            <param name="action_dispatch_topic" value="action_dispatch" />
            <param name="action_feedback_topic" value="action_feedback" />
            </node>
            
             <!-- ACTIONS -->

            <node name="rosplan_interface_start" pkg="cluedo_rosplan_interface" type="start" respawn="false" output="screen">
		<param name="knowledge_base" value="rosplan_knowledge_base" />
		<param name="pddl_action_name" value="start_game" />
		<param name="action_dispatch_topic" value="/rosplan_plan_dispatcher/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_dispatcher/action_feedback" />
            </node>
            
            <node name="rosplan_interface_leave_oracle" pkg="cluedo_rosplan_interface" type="leaveOracle" respawn="false" output="screen">
		<param name="knowledge_base" value="rosplan_knowledge_base" />
		<param name="pddl_action_name" value="leave_oracle" />
		<param name="action_dispatch_topic" value="/rosplan_plan_dispatcher/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_dispatcher/action_feedback" />
            </node>

             <node name="rosplan_interface_collect_hint" pkg="cluedo_rosplan_interface" type="collectHint" respawn="false" output="screen">
		<param name="knowledge_base" value="rosplan_knowledge_base" />
		<param name="pddl_action_name" value="collect_hint" />
		<param name="action_dispatch_topic" value="/rosplan_plan_dispatcher/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_dispatcher/action_feedback" />
            </node>

            <node name="rosplan_interface_consistency_query" pkg="cluedo_rosplan_interface" type="consistencyQuery" respawn="false" output="screen">
		<param name="knowledge_base" value="rosplan_knowledge_base" />
		<param name="pddl_action_name" value="consistency_query" />
		<param name="action_dispatch_topic" value="/rosplan_plan_dispatcher/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_dispatcher/action_feedback" />
            </node>

             <node name="rosplan_interface_go_to_next_point" pkg="cluedo_rosplan_interface" type="goToNextPoint" respawn="false" output="screen">
		<param name="knowledge_base" value="rosplan_knowledge_base" />
		<param name="pddl_action_name" value="go_to_next_point" />
		<param name="action_dispatch_topic" value="/rosplan_plan_dispatcher/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_dispatcher/action_feedback" />
            </node>

            <node name="rosplan_interface_go_to_oracle" pkg="cluedo_rosplan_interface" type="goToOracle" respawn="false" output="screen">
		<param name="knowledge_base" value="rosplan_knowledge_base" />
		<param name="pddl_action_name" value="go_to_oracle" />
		<param name="action_dispatch_topic" value="/rosplan_plan_dispatcher/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_dispatcher/action_feedback" />
            </node>

            <node name="rosplan_interface_solution_query" pkg="cluedo_rosplan_interface" type="solutionQuery" respawn="false" output="screen">
		<param name="knowledge_base" value="rosplan_knowledge_base" />
		<param name="pddl_action_name" value="solution_query" />
		<param name="action_dispatch_topic" value="/rosplan_plan_dispatcher/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_dispatcher/action_feedback" />
            </node>

            <!-- shell file to call rosplan services-->
            <node name="rosplan" pkg="erl2" type="rosplanLaunch.sh" />      
            
            <!-- Replanning node 
            <node name="replan" pkg="erl2" type ="replan.py"/>-->
            
            
            <!-- virtual ACTIONS         
           
                <arg name="pddl_action_name" />
	            <arg name="action_duration"		default="1.0" />
	            <arg name="action_probability"	default="1.0" />

                <include file="$(find rosplan_planning_system)/launch/includes/simulated_action.launch" >
                <arg name="pddl_action_name" value="start_game" />
                <param name="action_duration"		value="$(arg action_duration)" />
		        <param name="action_probability"	value="$(arg action_probability)" />
                </include>

                <include file="$(find rosplan_planning_system)/launch/includes/simulated_action.launch" >
                <arg name="pddl_action_name" value="leave_oracle" />
                <param name="action_duration"		value="$(arg action_duration)" />
		        <param name="action_probability"	value="$(arg action_probability)" />
                </include>

                <include file="$(find rosplan_planning_system)/launch/includes/simulated_action.launch" >
                <arg name="pddl_action_name" value="collect_hint" />
                <param name="action_duration"		value="$(arg action_duration)" />
		        <param name="action_probability"	value="$(arg action_probability)" />
                </include>

                <include file="$(find rosplan_planning_system)/launch/includes/simulated_action.launch" >
                <arg name="pddl_action_name" value="consistency_query" />
                <param name="action_duration"		value="$(arg action_duration)" />
		        <param name="action_probability"	value="1.0" />
                </include>

                 <include file="$(find rosplan_planning_system)/launch/includes/simulated_action.launch" >
                <arg name="pddl_action_name" value="go_to_oracle" />
                <param name="action_duration"		value="$(arg action_duration)" />
		        <param name="action_probability"	value="$(arg action_probability)" />
                </include>

                 <include file="$(find rosplan_planning_system)/launch/includes/simulated_action.launch" >
                <arg name="pddl_action_name" value="go_to_next_point" />
                <param name="action_duration"		value="$(arg action_duration)" />
		        <param name="action_probability"	value="$(arg action_probability)" />
                </include>

                 <include file="$(find rosplan_planning_system)/launch/includes/simulated_action.launch" >
                <arg name="pddl_action_name" value="solution_query" />
                <param name="action_duration"		value="$(arg action_duration)" />
		        <param name="action_probability"	value="$(arg action_probability)" /> 
                </include>-->  
               
                 
</launch>
